app/register/index.js

```diff
@@ -0,0 +1,65 @@
+const submitButton = document.getElementById('submit-button');
+const passwordElement = document.getElementById('password');
+
+const clearWarningMessages = () => {
+	const warningMessages = document
+		.getElementById('warning-message-container')
+		.getElementsByTagName('P');
+	const numberOfMessages = warningMessages.length;
+
+	for (let index = 0; index < numberOfMessages; index++) {
+		warningMessages[index].classList.add('hidden');
+	}
+};
+
+const clearInputValue = (event) => {
+	event.target.value = '';
+};
+
+const enableSubmitButton = () => {
+	submitButton.disabled = false;
+};
+
+const disableSubmitButton = () => {
+	submitButton.disabled = true;
+};
+
+const showWarningMessage = (warningMessageElement) => {
+	document.getElementById(warningMessageElement).classList.toggle('hidden');
+};
+
+const validatePasswordCharacters = (regex, warningType) => (event) => {
+	if (!regex.test(event.target.value)) {
+		showWarningMessage(warningType);
+		disableSubmitButton();
+	}
+};
+
+const validatePassword = () => {
+	const checkPassword = (event) => {
+		if (event.target.value.length < 8) {
+			showWarningMessage('length');
+			disableSubmitButton();
+		}
+
+		const regExCheckNumber = /([0-9])/;
+		validatePasswordCharacters(regExCheckNumber, 'number')(event);
+
+		const regExCheckLowercase = /([a-z])/;
+		validatePasswordCharacters(regExCheckLowercase, 'lowercase')(event);
+
+		const regExCheckUppercase = /([A-Z])/;
+		validatePasswordCharacters(regExCheckUppercase, 'uppercase')(event);
+	};
+
+	passwordElement.addEventListener('change', checkPassword);
+};
+
+const resetPasswordInput = () => {
+	passwordElement.addEventListener('focus', clearInputValue);
```

Comment:
Would this cause the password element to be cleared every time they click on the text box?

Reply:
(Honestly it sounds kind of useful because I always end up retyping the damn thing anyway)

Reply:
Yes, it does. I think it's a pain to have to delete the contents before you start typing again. :)